react/features/video-menu/components/AbstractDemoteToVisitorDialog.ts

```diff
@@ -0,0 +1,57 @@
+import { Component } from 'react';
+import { WithTranslation } from 'react-i18next';
+
+import { createRemoteVideoMenuButtonEvent } from '../../analytics/AnalyticsEvents';
+import { sendAnalytics } from '../../analytics/functions';
+import { IStore } from '../../app/types';
+import { demoteRequest } from '../../visitors/actions';
+
+interface IProps extends WithTranslation {
+
+    /**
+     * The Redux dispatch function.
+     */
+    dispatch: IStore['dispatch'];
+
+    /**
+     * The ID of the remote participant to be demoted.
+     */
+    participantID: string;
+}
+
+/**
+ * Abstract dialog to confirm a remote participant demote action.
+ */
+export default class AbstractDemoteToVisitorDialog
```

Comment:
Do we really need a whole abstract class component for this. If we share only the submit function maybe we can extract it in separate action and use it in both web and native?